client/state/wp-data/calypso-wp-data-provider.js

```diff
@@ -0,0 +1,40 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { use, RegistryProvider } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import calypsoRegistryPlugin from './calypso-registry-plugin';
+import calypsoWPDataStore from './calypso-wp-data-store';
+
+class CalypsoWPDataProvider extends Component {
+	constructor() {
+		super( ...arguments );
+		this.registry = undefined;
+	}
+
+	updateRegistry( calypsoStore ) {
+		this.registry = use( calypsoRegistryPlugin( calypsoStore ) );
+		this.registry.registerStore( 'calypso', calypsoWPDataStore );
+	}
+
+	componentDidMount() {
+		this.updateRegistry( this.props.calypsoStore );
```

Comment:
It's better to init the registry in the component constructor. Then it will get initialized in server side rendering, too. `componentDid{Mount,Update}` is never called in SSR and the registry will remain undefined.

Reply:
If we do that, then I'll need to make sure the `store` prop never changes. Is that what we want to do?

Reply:
Also, the `calypsoStore` prop doesn't exist when the constructor is run, so it would need to be stitched up afterwards.

Reply:
> If we do that, then I'll need to make sure the store prop never changes. Is that what we want to do?

The `componentDidUpdate` method will still be there to take care of the prop change. I'm suggesting only merging `componentDidMount` into the constructor. The constructor is the only one that gets called in both SSR and DOM rendering.

> Also, the calypsoStore prop doesn't exist when the constructor is run, so it would need to be stitched up afterwards.

That's strange. Are you sure it doesn't exist? Getting `this.props` initialized is the (only) reason why React component constructor often call `super( props )`:
```js
constructor( props ) {
  // this.props is undefined on this line
  super( props );
  // the super call just initialized them!
  this.registry = this.updateRegistry( this.calypsoStore );
}
```
Is you constructor doing the `super` call?

Reply:
I just checked again, and `calypsoStore` is in fact coming through on the constructor. I'll adjust this to remove `componentDidMount` then. Thanks!